Import backports during build

jakirkham opened the pull request:
Just as a safety measure, make sure to import `backports` during the build. This as much a test as a build step. Though as `conda-build` compiles all Python files for us, it is not strictly necessary as a build step. Still it is good to have in case `conda-build` ever changes its behavior.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Create __init__ with Python

Instead of using OS specific strategies to create `__init__.py`, simply
open the file for writing with Python and close it right after. This
should `backports` to be a noarch Python package in the future. Also
just simplifies the build steps as well.

jakirkham committed:
Switch back to the source directory

After having created the `backports` module, switch back to the source
directory. This is done in preparation for other build steps that may
assume they are being run in the source directory.

jakirkham committed:
Import backports during build

Just as a safety measure, make sure to import `backports` during the
build. This as much a test as a build step. Though as `conda-build`
compiles all Python files for us, it is not strictly necessary as a
build step. Still it is good to have in case `conda-build` ever changes
its behavior.

jakirkham committed:
Bump build number to 1

As we have altered the build by writing out the `__init__.py` with
Python and adding the import of `backports` into the build, bump the
build number. No real changes are expected with this build change.
However it is still good to demarcate the change should something come
up.
